src/jace/util/compiling.py

```diff
@@ -0,0 +1,146 @@
+# JaCe - JAX Just-In-Time compilation using DaCe (Data Centric Parallel Programming)
+#
+# Copyright (c) 2024, ETH Zurich
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Contains everything for compiling and running `TranslatedJaxprSDFG` instances."""
+
+from __future__ import annotations
+
+import time
+from typing import TYPE_CHECKING, Any
+
+import dace
+import numpy as np
+from dace import data as dace_data
+
+from jace import util
+
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping, Sequence
+
+    from jace import translator
+    from jace.util import dace_helper
+
+
+def compile_jax_sdfg(
+    tsdfg: translator.TranslatedJaxprSDFG,
+) -> dace_helper.CompiledSDFG:
+    """Compiles the SDFG embedded in `tsdfg` and return the resulting `CompiledSDFG` object.
+
+    Note:
+        For calling the returned `CompiledSDFG` object you need the `inp_names` and `out_names`
+        of the input `TranslatedJaxprSDFG`.
+    """
+    if any(  # We do not support the DaCe return mechanism
+        arrname.startswith("__return")
+        for arrname in tsdfg.sdfg.arrays.keys()  # noqa: SIM118  # we can not use `in` because we are also interested in `__return_`!
+    ):
+        raise ValueError("Only support SDFGs without '__return' members.")
+
+    # To ensure that the SDFG is compiled and to get rid of a warning we must modify
+    #  some settings of the SDFG. To fake an immutable SDFG, we will restore them later.
+    sdfg = tsdfg.sdfg
+    org_sdfg_name = sdfg.name
+    org_recompile = sdfg._recompile
+    org_regenerate_code = sdfg._regenerate_code
+
+    try:
+        # We need to give the SDFG another name, this is needed to prevent a DaCe error/warning.
+        #  This happens if we compile the same lowered SDFG multiple times with different options.
+        sdfg.name = f"{sdfg.name}__comp_{int(time.time() * 1000)}"
+
+        with dace.config.temporary_config():
+            sdfg._recompile = True
+            sdfg._regenerate_code = True
+            dace.Config.set("compiler", "use_cache", value=False)
+            csdfg: dace_helper.CompiledSDFG = sdfg.compile()
+
+    finally:
+        sdfg.name = org_sdfg_name
+        sdfg._recompile = org_recompile
+        sdfg._regenerate_code = org_regenerate_code
+
+    return csdfg
+
+
+def run_jax_sdfg(
+    csdfg: dace_helper.CompiledSDFG,
+    inp_names: Sequence[str],
+    out_names: Sequence[str],
+    cargs: Sequence[Any],
+    ckwargs: Mapping[str, Any],
```

Comment:
As mentioned in previous reviews, this naming style is not very pythonic and not very coherent with the other parameter names in this function. I think either `args` & `kwags` or `call_args` & `call_kwargs` would be more appropriate and standard.